crates/sc-consensus-subspace-rpc/src/lib.rs

```diff
@@ -764,14 +764,13 @@ where
             }
         };
 
-        let indices = archived_segment
-            .segment_header
-            .segment_index()
-            .segment_piece_indexes();
         let pieces = &archived_segment.pieces;
-        for (piece_index, piece) in indices.into_iter().zip(pieces.iter()) {
-            if requested_piece_index == piece_index {
-                return Ok(Some(piece.to_vec()));
+        if requested_piece_index.segment_index() == archived_segment.segment_header.segment_index()
+        {
+            if pieces.len() >= (requested_piece_index.position() + 1) as usize {
```

Comment:
This check is unnecessary, position is logically guaranteed to be within segment, just `Ok(Some(archived_segment.pieces[requested_piece_index.position() as usize].to_vec()))` is sufficient here.

Reply:
OK, I was not sure whether the segment would always be full.